gabbi/runner.py

```diff
@@ -89,26 +96,48 @@ def run():
              'form package.module or package.module:class.'
     )
 
-    args = parser.parse_args()
+    args = parser.parse_args(argv[1:])
     host, port, prefix, force_ssl = utils.host_info_from_target(
         args.target, args.prefix)
 
-    # Initialize response handlers.
-    handler_objects = initialize_handlers(args.response_handlers)
+    response_handlers = initialize_handlers(args.response_handlers)
 
-    data = utils.load_yaml(handle=sys.stdin)
+    failfast = args.failfast
+    failure = False
+    for input_file in input_files:
+        params = (response_handlers, host, port, prefix, force_ssl, failfast)
+        # XXX(FND): special-casing; use generic stream detection instead?
+        if input_file is sys.stdin:
+            success = execute(input_file, *params)
+        else:  # file path
+            with open(input_file, 'r') as fh:
+                success = execute(fh, *params)
+
+        if not failure:  # once failed, this is considered immutable
+            failure = not success
+        if failure and failfast:
+            break
+
+    sys.exit(failure)
+
+
+def execute(handle, response_handlers, host, port, prefix,  # TODO(FND): rename
+            force_ssl=False, failfast=False):
+    data = utils.load_yaml(handle)
     if force_ssl:
         if 'defaults' in data:
             data['defaults']['ssl'] = True
         else:
             data['defaults'] = {'ssl': True}
+
     loader = unittest.defaultTestLoader
     test_suite = suitemaker.test_suite_from_dict(
         loader, 'input', data, '.', host, port, None, None, prefix=prefix,
-        handlers=handler_objects)
+        handlers=response_handlers)
```

Comment:
doing `s/handler_objects/response_handlers/` on the file is an unrelated change that you seem to only be doing because you don't like/understand the name. Why is the change a part of this patchset?